Sources/TuistServer/Services/MultipartUploadArtifactService.swift

```diff
@@ -84,17 +85,29 @@ public final class MultipartUploadArtifactService: MultipartUploadArtifactServic
                 guard let url = URL(string: uploadURLString) else {
                     throw MultipartUploadArtifactServiceError.invalidMultipartUploadURL(uploadURLString)
                 }
-                let request = uploadRequest(url: url, fileSize: UInt64(bytesRead), data: partData)
-                let etag = try await upload(for: request)
-                parts.append((etag: etag, partNumber: Int(partNumber)))
+                uploadParts.append(
+                    UploadPart(
+                        url: url,
+                        fileSize: UInt64(bytesRead),
+                        data: partData,
```

Comment:
Is it ok memory-wise to hold this in memory?

Reply:
I believe so, yeah 👍 We won't hold too much here, max the size of the `.xcframework` or app being uploaded.